.github/workflows/push.yml

```diff
@@ -16,8 +16,45 @@ jobs:
       - name: Set up JDK
         uses: actions/setup-java@v4
         with:
-          distribution: 'temurin'
-          java-version: '21'
+          distribution: "temurin"
+          java-version: "21"
 
       - name: Run tests
         run: ./gradlew test
+
+  regenerate-deps:
+    permissions:
+      contents: write
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+
+      - name: Restore and save Nix Store
+        uses: nix-community/cache-nix-action@v6.1.3
+        with:
+          primary-key: nix-${{ runner.os }}-${{ hashFiles('**/*.nix', '**/flake.lock') }}
+          gc-max-store-size-linux: 1G
+          purge: true
+
+      - name: Update deps.json file for nix
+        run: |
+          cp $(nix build --no-link --print-out-paths .#kotlin-lsp.mitmCache.updateScript) update-deps.sh && chmod +x update-deps.sh
+          sed -i 's/^if \[ -n "\$useBwrap" \]; then$/if false; then/' update-deps.sh
+          ./update-deps.sh
+
+      - name: Commit deps.json file
+        run: |
+          if [ ! -n "$(git status --porcelain)" ]; then
+            exit
+          fi
+          git add nix/deps.json
+          git config user.name "github-actions"
+          git config user.email "action@github.com"
+          git commit -m "build(nix): update deps.json [skip ci]"
+          git push
```

Comment:
I'd like this to create a PR instead of pushing directly to main, I'm planning to use dependabot this way as well